src/Writers/HTMLWriter.jl

```diff
@@ -117,21 +144,60 @@ Documenter.HTML(assets = [
     # Standard remote asset (extension used to determine that class = :js)
     asset("https://example.com/jslibrary.js"),
     # Setting asset class manually, since it can't be determined manually
-    asset("https://example.com/fonts", class = :css),
+    asset("https://example.com/fonts", class=:css),
     # Same as above, but for a local asset
     asset("asset/foo.script", class=:js, islocal=true),
+    # Asset provided via source and destination path
+    asset("/abs/path/to/extra_style.css", dst="asset/extra_style.css")
 ])
 ```
 """
-function asset(uri; class = nothing, islocal=false)
+function asset(src::String; class = nothing, islocal = nothing, dst = "")
```

Comment:
I somehow can't befriend me with functions without type restrictions and default values of nothing...
Sure in some cases it is useful to not restrict the types but it leads to more work. As now one has to test for valid input types and valid arguments on top. Isn't that one of the reasons to use types to have the function refuse obviously wrong arguments?
In the case of the optional arguments here i can understand it as they might be derived of the `src<:AbstractString`.

Reply:
Ah, but in this case, `nothing` indicates a missing `dst` value, i.e. that the destination path is not set, instead of a simple default path. To me, using an empty string to indicate a missing value is similar to using `-1` for that if you know that otherwise you're expecting a positive number --- feels ugly in an age where type unions are cheap. :slightly_smiling_face: In any case, thanks for changing it!